{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "1901f0ea-c6ec-4f8a-9a4c-6d9b59162fe6",
   "metadata": {},
   "source": [
    "# Ad-Hoc Analyses\n",
    "- **Project:** GP2 AFR-AAC meta-GWAS \n",
    "- **Version:** Mix of bash and R\n",
    "- **Status:** COMPLETE\n",
    "- **Started:** 02-MAY-2023\n",
    "- **Last Updated:** 02-MAY-2023\n",
    "    - **Update Description:** Adding ad-hoc analyses to one notebook\n",
    "\n",
    "## Notebook Overview\n",
    "- Investigating runs of homozygosity, linear regression for % admixture and age, haplotypes with 1KG, fine-mapping, dominant vs recessive modeling, annotating WGS for T2, conditional analysis, Miami plot compared to Europeans, and GBA Gauchian caller WDL\n",
    "\n",
    "### CHANGELOG\n",
    "- 02-MAY-2023: Notebook started \n",
    "\n",
    "---\n",
    "# Data Overview \n",
    "\n",
    "| ANCESTRY |     DATASET     | CASES | CONTROLS |  TOTAL  |           ARRAY           |                NOTES                |\n",
    "|:--------:|:---------------:|:-----:|:--------:|:-------------------------:|:---------------------------------------------------------------------------------------------------------------:|:-----------------------------------:|\n",
    "|    AFR   | IPDGC – Nigeria |  304  |    285   |   589   |         NeuroChip         | . | \n",
    "|    AFR   |  GP2  |  711  |   1,011  |  1,722  |        NeuroBooster       | . |\n",
    "|    AAC   |  GP2 |  185  |   1,149  |  1,334  |        NeuroBooster       | . | \n",
    "|    AAC   |     23andMe     |  288  |  193,985 | 194,273 | Omni Express & GSA & 550k |        Just summary statistics       |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d4e293f8-199a-441c-91a5-1b22cc10ab1a",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b18c9f00-93c5-462d-8fbd-7dfe20a0ae1c",
   "metadata": {},
   "source": [
    "# Runs of Homozygosity"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "52bd98b1-e160-4e18-9587-4bd0f8481e23",
   "metadata": {},
   "source": [
    "```bash\n",
    "## ROH in GP2 AFR - NeuroBooster data\n",
    "\n",
    "mkdir ROH\n",
    "\n",
    "plink --bfile UPDATED-GP2-v3-AFR-wNIGERIAN-NB \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out ROH/pruning\n",
    "\n",
    "cd ROH\n",
    "\n",
    "plink --bfile UPDATED-GP2-v3-AFR-wNIGERIAN-NB \\\n",
    "--extract pruning.prune.in \\\n",
    "--make-bed --out pruned_data\n",
    "\n",
    "plink --bfile pruned_data \\\n",
    "--homozyg \\\n",
    "--homozyg-density 50 \\\n",
    "--homozyg-gap 1000 \\\n",
    "--homozyg-kb 1000 \\\n",
    "--homozyg-snp 10 \\\n",
    "--homozyg-window-het 1 \\\n",
    "--homozyg-window-missing 10 \\\n",
    "--homozyg-window-snp 20 \\\n",
    "--homozyg-window-threshold 0.05 \\\n",
    "--out pruned_ROH_UPDATED-GP2-v3-AFR-wNIGERIAN-NB\n",
    "\n",
    "# only chr 1\n",
    "plink --bfile pruned_data \\\n",
    "--chr 1 \\\n",
    "--homozyg \\\n",
    "--homozyg-density 50 \\\n",
    "--homozyg-gap 1000 \\\n",
    "--homozyg-kb 1000 \\\n",
    "--homozyg-snp 10 \\\n",
    "--homozyg-window-het 1 \\\n",
    "--homozyg-window-missing 10 \\\n",
    "--homozyg-window-snp 20 \\\n",
    "--homozyg-window-threshold 0.05 \\\n",
    "--out pruned_ROH_UPDATED-GP2-v3-AFR-wNIGERIAN-NB_chr1\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a76d8d49-cf1b-45ab-879c-ec4fa85ec7ca",
   "metadata": {},
   "source": [
    "```R\n",
    "## Explore NSEG, KB and KBAVG differences in cases versus controls\n",
    "data <- read.table(\"pruned_ROH_UPDATED-GP2-v3-AFR-wNIGERIAN-NB.hom.indiv\", header = T)\n",
    "head(data)\n",
    "data$case <- data$PHE - 1\n",
    "model1 <- glm(case ~ NSEG, data = data, family = \"binomial\")\n",
    "summary(data$case)\n",
    "dat <- subset(data, case != -10)\n",
    "length(dat$case)\n",
    "model1 <- glm(case ~ NSEG, data = dat, family = \"binomial\")\n",
    "summary(model1)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "347606c0-eaff-421f-be29-314fcb542cf9",
   "metadata": {},
   "source": [
    "```\n",
    "# Call:\n",
    "# glm(formula = case ~ NSEG, family = \"binomial\", data = dat)\n",
    "# \n",
    "# Deviance Residuals: \n",
    "#     Min       1Q   Median       3Q      Max  \n",
    "# -2.4855  -1.0001  -0.8484   1.2944   1.6926  \n",
    "# \n",
    "# Coefficients:\n",
    "#             Estimate Std. Error z value Pr(>|z|)    \n",
    "# (Intercept) -2.12945    0.23385  -9.106  < 2e-16 ***\n",
    "# NSEG         0.08081    0.01052   7.684 1.54e-14 ***\n",
    "# ---\n",
    "# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "# \n",
    "# (Dispersion parameter for binomial family taken to be 1)\n",
    "# \n",
    "#     Null deviance: 2300.5  on 1701  degrees of freedom\n",
    "# Residual deviance: 2232.0  on 1700  degrees of freedom\n",
    "# AIC: 2236\n",
    "# \n",
    "# Number of Fisher Scoring iterations: 4\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53480e6f-c7a9-44fa-82e5-3ec1bcd4f08f",
   "metadata": {},
   "source": [
    "```R\n",
    "model2 <- glm(case ~ KB, data = dat, family = \"binomial\")\n",
    "summary(model2)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0f27680-3ab0-40b3-b717-17d486ef3dd9",
   "metadata": {},
   "source": [
    "```\n",
    "# Call:\n",
    "# glm(formula = case ~ KB, family = \"binomial\", data = dat)\n",
    "# \n",
    "# Deviance Residuals: \n",
    "#     Min       1Q   Median       3Q      Max  \n",
    "# -2.6162  -1.0014  -0.9498   1.3290   1.4806  \n",
    "# \n",
    "# Coefficients:\n",
    "#               Estimate Std. Error z value Pr(>|z|)    \n",
    "# (Intercept) -1.006e+00  1.444e-01  -6.969 3.19e-12 ***\n",
    "# KB           1.593e-05  3.466e-06   4.596 4.31e-06 ***\n",
    "# ---\n",
    "# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "# \n",
    "# (Dispersion parameter for binomial family taken to be 1)\n",
    "# \n",
    "#     Null deviance: 2300.5  on 1701  degrees of freedom\n",
    "# Residual deviance: 2271.6  on 1700  degrees of freedom\n",
    "# AIC: 2275.6\n",
    "# \n",
    "# Number of Fisher Scoring iterations: 4\n",
    "\n",
    "# model3 <- glm(case ~ KBAVG, data = dat, family = \"binomial\")\n",
    "# summary(model3)\n",
    "# \n",
    "# Call:\n",
    "# glm(formula = case ~ KBAVG, family = \"binomial\", data = dat)\n",
    "# \n",
    "# Deviance Residuals: \n",
    "#     Min       1Q   Median       3Q      Max  \n",
    "# -1.4197  -1.0174  -0.9996   1.3410   1.3939  \n",
    "# \n",
    "# Coefficients:\n",
    "#               Estimate Std. Error z value Pr(>|z|)   \n",
    "# (Intercept) -0.8328821  0.2534119  -3.287  0.00101 **\n",
    "# KBAVG        0.0002524  0.0001372   1.840  0.06573 . \n",
    "# ---\n",
    "# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "# \n",
    "# (Dispersion parameter for binomial family taken to be 1)\n",
    "# \n",
    "#     Null deviance: 2300.5  on 1701  degrees of freedom\n",
    "# Residual deviance: 2297.0  on 1700  degrees of freedom\n",
    "# AIC: 2301\n",
    "# \n",
    "# Number of Fisher Scoring iterations: 4\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3360024-e1d5-4f98-876a-6339fa9324f8",
   "metadata": {},
   "source": [
    "```bash\n",
    "## ROH in GP2 AAC - Neurobooster data\n",
    "cd /data/GP2/projects/2023_01_MBM_AFR_AAC_GWAS/data/AAC/\n",
    "\n",
    "mkdir ROH\n",
    "\n",
    "plink --bfile UPDATED-GP2-v3-AAC/UPDATED-GP2-v3-AAC \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out ROH/pruning\n",
    "\n",
    "cd ROH\n",
    "\n",
    "plink --bfile /data/GP2/projects/2023_01_MBM_AFR_AAC_GWAS/data/AAC/UPDATED-GP2-v3-AAC/UPDATED-GP2-v3-AAC \\\n",
    "--extract pruning.prune.in \\\n",
    "--make-bed --out pruned_data\n",
    "\n",
    "plink --bfile pruned_data \\\n",
    "--homozyg \\\n",
    "--homozyg-density 50 \\\n",
    "--homozyg-gap 1000 \\\n",
    "--homozyg-kb 1000 \\\n",
    "--homozyg-snp 10 \\\n",
    "--homozyg-window-het 1 \\\n",
    "--homozyg-window-missing 10 \\\n",
    "--homozyg-window-snp 20 \\\n",
    "--homozyg-window-threshold 0.05 \\\n",
    "--out pruned_ROH_UPDATED-GP2-v3-AAC\n",
    "\n",
    "# only chr 1\n",
    "plink --bfile pruned_data \\\n",
    "--chr 1 \\\n",
    "--homozyg \\\n",
    "--homozyg-density 50 \\\n",
    "--homozyg-gap 1000 \\\n",
    "--homozyg-kb 1000 \\\n",
    "--homozyg-snp 10 \\\n",
    "--homozyg-window-het 1 \\\n",
    "--homozyg-window-missing 10 \\\n",
    "--homozyg-window-snp 20 \\\n",
    "--homozyg-window-threshold 0.05 \\\n",
    "--out pruned_ROH_UPDATED-GP2-v3-AAC_chr1\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "265f5312-d6ea-4579-a6df-e6f4b4535e92",
   "metadata": {},
   "source": [
    "```R\n",
    "## Explore NSEG, KB and KBAVG differences in cases versus controls\n",
    "data <- read.table(\"pruned_ROH_UPDATED-GP2-v3-AAC.hom.indiv\", header = T)\n",
    "head(data)\n",
    "data$case <- data$PHE - 1\n",
    "dat <- subset(data, case != -10)\n",
    "model1 <- glm(case ~ NSEG, data = dat, family = \"binomial\")\n",
    "summary(model1)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a20cdb16-ac92-48e7-8743-33f1680d37f5",
   "metadata": {},
   "source": [
    "```\n",
    "# Call:\n",
    "# glm(formula = case ~ NSEG, family = \"binomial\", data = dat)\n",
    "# \n",
    "# Deviance Residuals: \n",
    "#     Min       1Q   Median       3Q      Max  \n",
    "# -0.7022  -0.5442  -0.5363  -0.5284   2.0326  \n",
    "# \n",
    "# Coefficients:\n",
    "#             Estimate Std. Error z value Pr(>|z|)    \n",
    "# (Intercept) -2.02535    0.26761  -7.568 3.78e-14 ***\n",
    "# NSEG         0.01058    0.01504   0.703    0.482    \n",
    "# ---\n",
    "# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "# \n",
    "# (Dispersion parameter for binomial family taken to be 1)\n",
    "# \n",
    "#     Null deviance: 1046.1  on 1313  degrees of freedom\n",
    "# Residual deviance: 1045.6  on 1312  degrees of freedom\n",
    "# AIC: 1049.6\n",
    "# \n",
    "# Number of Fisher Scoring iterations: 4\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7584f1d8-c968-4807-a477-4b66c60883b5",
   "metadata": {},
   "source": [
    "```R\n",
    "model2 <- glm(case ~ KB, data = dat, family = \"binomial\")\n",
    "summary(model2)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56b7ed92-d402-4672-9d2d-cd197e884cdf",
   "metadata": {},
   "source": [
    "```\n",
    "# Call:\n",
    "# glm(formula = case ~ KB, family = \"binomial\", data = dat)\n",
    "# \n",
    "# Deviance Residuals: \n",
    "#     Min       1Q   Median       3Q      Max  \n",
    "# -0.8099  -0.5408  -0.5370  -0.5335   2.0200  \n",
    "# \n",
    "# Coefficients:\n",
    "#               Estimate Std. Error z value Pr(>|z|)    \n",
    "# (Intercept) -1.948e+00  1.252e-01 -15.559   <2e-16 ***\n",
    "# KB           3.178e-06  2.949e-06   1.078    0.281    \n",
    "# ---\n",
    "# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "# \n",
    "# (Dispersion parameter for binomial family taken to be 1)\n",
    "# \n",
    "#     Null deviance: 1046.1  on 1313  degrees of freedom\n",
    "# Residual deviance: 1045.0  on 1312  degrees of freedom\n",
    "# AIC: 1049\n",
    "# \n",
    "# Number of Fisher Scoring iterations: 4\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ccc14e0-bc57-463b-aa05-1a9c5e7eee04",
   "metadata": {},
   "source": [
    "```R\n",
    "model3 <- glm(case ~ KBAVG, data = dat, family = \"binomial\")\n",
    "summary(model3)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48181b57-5ee8-44da-94fc-371222ca3701",
   "metadata": {},
   "source": [
    "```\n",
    "# Call:\n",
    "# glm(formula = case ~ KBAVG, family = \"binomial\", data = dat)\n",
    "# \n",
    "# Deviance Residuals: \n",
    "#     Min       1Q   Median       3Q      Max  \n",
    "# -0.7762  -0.5415  -0.5369  -0.5319   2.0324  \n",
    "# \n",
    "# Coefficients:\n",
    "#               Estimate Std. Error z value Pr(>|z|)    \n",
    "# (Intercept) -2.1132342  0.2782441  -7.595 3.08e-14 ***\n",
    "# KBAVG        0.0001456  0.0001447   1.006    0.314    \n",
    "# ---\n",
    "# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "# \n",
    "# (Dispersion parameter for binomial family taken to be 1)\n",
    "# \n",
    "#     Null deviance: 1046.1  on 1313  degrees of freedom\n",
    "# Residual deviance: 1045.2  on 1312  degrees of freedom\n",
    "# AIC: 1049.2\n",
    "# \n",
    "# Number of Fisher Scoring iterations: 4\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b54686f5-d66e-4cd2-843f-4e865e4bea78",
   "metadata": {},
   "source": [
    "# Linear Regression\n",
    "Age and % admixture"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "4741d944-7543-42a2-89f4-a77f7f448183",
   "metadata": {},
   "source": [
    "```bash\n",
    "######## LINEAR REGRESSION VERSUS AGE ########\n",
    "\n",
    "### AFR\n",
    "cd GP2-v4-AFR-wNIGERIAN-NB/\n",
    "\n",
    "# Create age file\n",
    "awk '{print $1, $2, $20}' masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt > age_cases.txt\n",
    "\n",
    "# Filter cases (711 AFR)\n",
    "plink --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--filter-cases \\\n",
    "--make-bed --out Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds\n",
    "\n",
    "# Extract top hit and run linear regression \n",
    "plink --bfile Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--linear --pheno age_cases.txt --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt --covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --out linear_regression\n",
    "\n",
    "grep \"ADD\" linear_regression.assoc.linear > results_linear_regression_age_AFR.txt\n",
    "# (BETA =-2.004, STAT = -3.483, p = 0.0005284)\n",
    "# BETA/STAT = SE\n",
    "\n",
    "### AAC\n",
    "cd AAC/GP2-v4-AAC\n",
    "\n",
    "# Filter cases (185 AAC)\n",
    "plink --bfile GP2-v4-AAC-noRelateds --filter-cases --out Cases_GP2-v4-AAC-noRelateds --make-bed\n",
    "\n",
    "# Extract top hit and run linear regression (only cases)\n",
    "plink --bfile Cases_GP2-v4-AAC-noRelateds --linear --pheno AFR/GP2-v4-AFR-wNIGERIAN-NB/age_cases.txt --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt --covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --out linear_regression\n",
    "\n",
    "grep \"ADD\" linear_regression.assoc.linear > results_linear_regression_age_AAC.txt\n",
    "# (BETA =-4.156 , STAT = -2.443, p = 0.01559)\n",
    "# BETA/STAT = SE\n",
    "\n",
    "# Extract HMZ and HETs and run linear regression per genotype\n",
    "plink --bfile Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds --chr 1 --from-bp 155235878 --to-bp 155235878 --recode A --out Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds\n",
    "\n",
    "# HMZ for the risk allele versus HMZ for the non-risk allele \n",
    "plink --bfile Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds --remove HET_risk.txt --linear --pheno age_cases.txt --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt --covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --out linear_regression_HMZ_risk\n",
    "#  CHR                  SNP         BP   A1       TEST    NMISS       BETA         STAT            P \n",
    "#    1   chr1:155235878:G:T  155235878    G        ADD      409      -1.96       -3.062     0.002351\n",
    "\n",
    "# HET for the risk allele versus HMZ for the non-risk allele \n",
    "plink --bfile Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds --remove HMZ_risk.txt --linear --pheno age_cases.txt --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt --covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --out linear_regression_HET\n",
    "# CHR                  SNP         BP   A1       TEST    NMISS       BETA         STAT            P \n",
    "#    1   chr1:155235878:G:T  155235878    G        ADD      595     -2.285       -2.679     0.007585\n",
    "\n",
    "# HET for the risk allele versus HMZ for risk allele \n",
    "plink --bfile Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds --remove HMZ_non_risk.txt --linear --pheno age_cases.txt --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt --covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --out linear_regression_HET\n",
    "#  CHR                  SNP         BP   A1       TEST    NMISS       BETA         STAT            P \n",
    "#    1   chr1:155235878:G:T  155235878    T        ADD      370      1.455        1.145       0.2529\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b53d93a-047e-47c8-9c78-d096d8e78188",
   "metadata": {},
   "source": [
    "```R\n",
    "## Data visualization\n",
    "\n",
    "library(data.table)\n",
    "library(ggplot2)\n",
    "genotypes <- fread(\"GP2-v4-AFR-wNIGERIAN-NB-noRelateds.rs3115534.raw\", header =T) ## here we rename chr1* variant header to \"SNP\"\n",
    "covfile <- fread(\"masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt\", header =T)\n",
    "data <- merge(genotypes, covfile, by=\"IID\")\n",
    "data$SNP[data$SNP ==0] <- \"TT\"\n",
    "data$SNP[data$SNP==1] <- \"GT\"\n",
    "data$SNP[data$SNP ==2] <- \"GG\"\n",
    "\n",
    "dat <- subset(data, !is.na(data$SNP))\n",
    "data <- dat\n",
    "\n",
    "p <- ggplot(data, aes(x= SNP, y=AAO, fill=as.factor(data$SNP))) + geom_violin(trim=FALSE)\n",
    "p2 <- p+geom_boxplot(width=0.4, fill=\"white\" ) + theme_minimal()\n",
    "p2 + scale_fill_manual(values=c(\"lightblue\", \"orange\", \"green\")) + theme_bw() + ylab(\"Age\") +xlab(\"Genotypes\") + theme(legend.position = \"none\")\n",
    "ggsave(\"PLOT.jpeg\", dpi = 600, units = \"in\", height = 6, width = 6)\n",
    "```"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "a010916d-d8fe-4eb1-8a77-78cf86258417",
   "metadata": {},
   "source": [
    "```bash         \n",
    "######## MAF UPDATES ########\n",
    "\n",
    "plink --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds --chr 1 --from-bp 155235878 --to-bp 155235878 --freq --out freqs_AFR\n",
    "#  CHR                  SNP   A1   A2          MAF  NCHROBS\n",
    "#   1   chr1:155235878:G:T    G    T       0.2552     3346\n",
    "   \n",
    "plink --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds --chr 1 --from-bp 155235878 --to-bp 155235878 --assoc --out freqs_AFR\n",
    "# CHR                  SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR \n",
    "#   1   chr1:155235878:G:T  155235878    G   0.3362   0.1988    T         80.5    2.912e-19        2.042 \n",
    "   \n",
    "plink --bfile GP2-v4-AAC-noRelateds --chr 1 --from-bp 155235878 --to-bp 155235878 --freq --out freqs_AAC\n",
    "#  CHR                  SNP   A1   A2          MAF  NCHROBS\n",
    "#  1   chr1:155235878:G:T    G    T       0.1485     2646\n",
    "\n",
    "plink --bfile GP2-v4-AAC-noRelateds --chr 1 --from-bp 155235878 --to-bp 155235878 --assoc --out freqs_AAC\n",
    "# CHR                  SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR \n",
    "#   1   chr1:155235878:G:T  155235878    G   0.2268    0.136    T        20.57    5.764e-06        1.864 \n",
    "\n",
    "\n",
    "######## GENOME-WIDE ADMIXTURE AFR PERCENTAGES VERSUS RISK ########\n",
    "\n",
    "awk '{print $1, $2, $10}' GP2_merge_JAN_24_2023_ready_genotools_callrate_sex_ancestry_merge_train.10.Q.labeled > AFR_AAC_admixture\n",
    "\n",
    "# ALL cases and controls - looking at FID IID AFR",
    "plink --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--linear \\\n",
    "--pheno AFR_AAC_admixture \\\n",
    "--chr 1 --from-bp 155235878 --to-bp 155235878 \\\n",
    "--covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX --out admixture\n",
    "\n",
    "grep \"ADD\" admixture.assoc.linear > results_admixture.assoc.linear\n",
    "#(BETA = 0.0385, STAT=6.031, P=2.002E-9)\n",
    "# BETA/STAT = SE\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34934b8a-cd34-48bf-9bdd-63b8743a395d",
   "metadata": {},
   "source": [
    "# Haplotypes with 1KG\n",
    "\n",
    "SF9 using online 1KG website estimates, but this is an example"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "ec542269-98b3-42dd-9c55-ddd033d7a386",
   "metadata": {},
   "source": [
    "```bash\n",
    "## Calculate Haplotype blocks across AFR subpops using 1000 Genomes data\n",
    "\n",
    "cd AFR_AAC_NeuroBooster_Chr1\n",
    "\n",
    "## Packages \n",
    "module load plink/1.9\n",
    "\n",
    "## Keep list of individuals per subpop and generate subsetted plink files (only chr1)\n",
    "plink --bfile 1kg_30x_GRCh38_chr1 --keep SierraLeone.txt --make-bed --out SIERRALEONE\n",
    "plink --bfile 1kg_30x_GRCh38_chr1 --keep Esan.txt --make-bed --out ESAN\n",
    "plink --bfile 1kg_30x_GRCh38_chr1 --keep Gambian.txt --make-bed --out GAMBIAN\n",
    "plink --bfile 1kg_30x_GRCh38_chr1 --keep Yri.txt --make-bed --out YRI\n",
    "plink --bfile 1kg_30x_GRCh38_chr1 --keep Kenya.txt --make-bed --out KENYA\n",
    "\n",
    "## Calculate Hap blocks\n",
    "plink --bfile ESAN --blocks no-pheno-req --out ESAN\n",
    "plink --bfile GAMBIAN --blocks no-pheno-req --out GAMBIAN\n",
    "plink --bfile SIERRALEONE --blocks no-pheno-req --out SIERRALEONE\n",
    "plink --bfile YRI --blocks no-pheno-req --out YRI\n",
    "plink --bfile KENYA --blocks no-pheno-req --out KENYA\n",
    "\n",
    "## Grep Hap spanning risk signal\n",
    "grep \"chr1:155235878\" SIERRALEONE.blocks.det > SL\n",
    "grep \"chr1:155235878\" ESAN.blocks.det > ESAN\n",
    "grep \"chr1:155235878\" GAMBIAN.blocks.det > GAMBIAN\n",
    "grep \"chr1:155235878\" YRI.blocks.det > YRI\n",
    "grep \"chr1:155235878\" KENYA.blocks.det > KENYA\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eedf454b-7494-44a7-9934-ea3decc567f3",
   "metadata": {},
   "source": [
    "```\n",
    "# cat SL\n",
    "#    1    155235878    155247644       11.767     22 chr1:155235878:G:T|chr1:155236550:T:G|chr1:155237031:AT:A|chr1:155237758:T:C|chr1:155237835:G:A|chr1:155237942:C:T|chr1:155238055:T:C|chr1:155238392:T:C|chr1:155239060:T:C|chr1:155239288:G:A|chr1:155239569:C:T|chr1:155240239:G:A|chr1:155240707:T:C|chr1:155240850:A:C|chr1:155242309:A:G|chr1:155242420:A:G|chr1:155242497:C:T|chr1:155243333:T:A|chr1:155244275:C:T|chr1:155244682:A:G|chr1:155245686:C:T|chr1:155247644:AGG:A\n",
    "\n",
    "# cat ESAN\n",
    "#    1    155228473    155252386       23.914     51 chr1:155228473:CG:C|chr1:155229524:C:G|chr1:155229875:T:C|chr1:155229883:A:G|chr1:155230420:G:A|chr1:155230533:G:T|chr1:155230588:A:G|chr1:155231273:C:T|chr1:155231399:G:A|chr1:155231444:G:A|chr1:155231912:A:G|chr1:155232202:A:G|chr1:155232898:A:C|chr1:155233269:G:A|chr1:155234893:A:G|chr1:155235540:T:C|chr1:155235878:G:T|chr1:155236550:T:G|chr1:155237031:AT:A|chr1:155237758:T:C|chr1:155237835:G:A|chr1:155237942:C:T|chr1:155238055:T:C|chr1:155238392:T:C|chr1:155239060:T:C|chr1:155239288:G:A|chr1:155239569:C:T|chr1:155240239:G:A|chr1:155240707:T:C|chr1:155240850:A:C|chr1:155242309:A:G|chr1:155242420:A:G|chr1:155242497:C:T|chr1:155243333:T:A|chr1:155244275:C:T|chr1:155244682:A:G|chr1:155245686:C:T|chr1:155247644:AGG:A|chr1:155247647:A:ATT|chr1:155248283:G:A|chr1:155248377:T:C|chr1:155248574:T:C|chr1:155248655:A:G|chr1:155248887:C:G|chr1:155249416:A:T|chr1:155249834:C:A|chr1:155250023:A:G|chr1:155250077:C:A|chr1:155250974:A:G|chr1:155252046:C:A|chr1:155252386:C:T\n",
    "\n",
    "# cat GAMBIAN\n",
    "#    1    155235540    155237835        2.296      6 chr1:155235540:T:C|chr1:155235878:G:T|chr1:155236550:T:G|chr1:155237031:AT:A|chr1:155237758:T:C|chr1:155237835:G:A\n",
    "\n",
    "# cat YRI\n",
    "#    1    155216938    155258255       41.318     74 chr1:155216938:C:T|chr1:155216951:G:A|chr1:155220463:T:A|chr1:155223032:G:T|chr1:155223741:C:G|chr1:155224349:CAT:C|chr1:155225189:T:C|chr1:155225424:C:G|chr1:155226926:A:G|chr1:155227671:C:T|chr1:155227811:G:C|chr1:155228431:A:G|chr1:155228473:CG:C|chr1:155229524:C:G|chr1:155229875:T:C|chr1:155229883:A:G|chr1:155230420:G:A|chr1:155230533:G:T|chr1:155230588:A:G|chr1:155231273:C:T|chr1:155231399:G:A|chr1:155231444:G:A|chr1:155231912:A:G|chr1:155232202:A:G|chr1:155232898:A:C|chr1:155233269:G:A|chr1:155234893:A:G|chr1:155235540:T:C|chr1:155235878:G:T|chr1:155236550:T:G|chr1:155237031:AT:A|chr1:155237758:T:C|chr1:155237835:G:A|chr1:155237942:C:T|chr1:155238055:T:C|chr1:155238392:T:C|chr1:155239288:G:A|chr1:155239569:C:T|chr1:155240239:G:A|chr1:155240707:T:C|chr1:155240850:A:C|chr1:155242309:A:G|chr1:155242420:A:G|chr1:155242497:C:T|chr1:155243333:T:A|chr1:155244275:C:T|chr1:155244682:A:G|chr1:155245686:C:T|chr1:155247644:AGG:A|chr1:155247647:A:ATT|chr1:155248377:T:C|chr1:155248574:T:C|chr1:155248655:A:G|chr1:155248887:C:G|chr1:155249416:A:T|chr1:155249834:C:A|chr1:155250077:C:A|chr1:155250974:A:G|chr1:155252046:C:A|chr1:155252386:C:T|chr1:155252681:A:T|chr1:155252938:T:C|chr1:155252939:G:A|chr1:155253033:G:A|chr1:155253216:T:A|chr1:155253320:C:T|chr1:155254187:A:G|chr1:155254300:C:A|chr1:155254510:C:A|chr1:155255038:G:C|chr1:155255406:C:G|chr1:155256442:T:C|chr1:155258235:G:A|chr1:155258255:A:G\n",
    "\n",
    "# cat KENYA\n",
    "#    1    155235540    155243333        7.794     18 chr1:155235540:T:C|chr1:155235878:G:T|chr1:155236550:T:G|chr1:155237031:AT:A|chr1:155237758:T:C|chr1:155237835:G:A|chr1:155237942:C:T|chr1:155238055:T:C|chr1:155238392:T:C|chr1:155239288:G:A|chr1:155239569:C:T|chr1:155240239:G:A|chr1:155240707:T:C|chr1:155240850:A:C|chr1:155242309:A:G|chr1:155242420:A:G|chr1:155242497:C:T|chr1:155243333:T:A\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a4b12bc-e939-4ef6-b4ff-a57cf41fc2a0",
   "metadata": {},
   "source": [
    "# Fine-mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "190cc3c4-6812-4d37-bef1-813182c9de59",
   "metadata": {},
   "source": [
    "```R\n",
    "library(\"data.table\")\n",
    "install.packages(\"coloc\") \n",
    "source(\"https://bioconductor.org/biocLite.R\")\n",
    "biocLite(\"snpStats\")\n",
    "install.packages(\"robustbase\")\n",
    "library(\"robustbase\")\n",
    "install.packages(\"tidyr\")\n",
    "library(ggplot2)\n",
    "library(tidyr)\n",
    "devtools::install_github(\"chr1swallace/coloc\")\n",
    "library(\"coloc\")\n",
    "library(\"tidyverse\")\n",
    "\n",
    "finemap.abf(dataset, p1 = 1e-04)\n",
    "\n",
    "dataset1 <- fread(\"GBA_variants.tab\", header = T, sep = \"\\t\")\n",
    "dataset1$type <- \"cc\"\n",
    "dataset1$s <- 0.007\n",
    "dataset1$N <- 196054\n",
    "dataset_final<- dataset1 %>% mutate(StdErr_squared = StdErr^2)\n",
    "output <- dataset_final[,c(\"MarkerName\",\"Effect\",\"P-value\",\"StdErr_squared\", \"type\", \"s\", \"N\")]\n",
    "as.numeric(dataset_final$StdErr_squared)\n",
    "colnames(output) <- c(\"SNP\",\"beta\",\"P\",\"varbeta\", \"type\", \"s\", \"N\")\n",
    "fwrite(output, file = \"GBA.csv\", na = \"NA\", quote = F, row.names = F, sep = \"\\t\")\n",
    "results <- finemap.abf(dataset=list(SNP=output$SNP, beta=output$beta, varbeta=output$varbeta, N=196054,s=0.007,type=\"cc\"))\n",
    "combo <- cbind(results[1:5226,], output)\n",
    "hits <- subset(combo, SNP.PP > 0.2)\n",
    "fwrite(combo, file = \"GBA_results_fine_map.csv\", na = \"NA\", quote = F, row.names = F, sep = \",\")\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "580baeeb-e43d-49c4-b398-da2449fe5161",
   "metadata": {},
   "source": [
    "# Dominant vs recessive modeling"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "94a31f29-d1fe-4d29-90a3-e3a89fbaf543",
   "metadata": {},
   "source": [
    "```bash\n",
    "### 1_155235878 dominant model to test G-association in the AAC and AFR cohorts\n",
    "\n",
    "####### AAC #######\n",
    "cd GP2-v4-AAC\n",
    "\n",
    "plink2 --bfile GP2-v4-AAC-noRelateds \\\n",
    "--glm dominant --chr 1 --from-bp 155235878 --to-bp 155235878 \\\n",
    "--covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--out dominant_AAC --ci 0.95\n",
    "\n",
    "# CHROM\tPOS\tID\tREF\tALT\tA1\tTEST\tOBS_CT\tOR\tSE\tL95\tU95\tT_STAT\tP\n",
    "# 1\t155235878\tchr1:155235878:G:T\tT\tG\tG\tDOM\t1323\t1.96795\t0.171179\t1.40704\t2.75247\t3.9549\t7.65654e-05\n",
    "\n",
    "cd GP2-v4-AAC\n",
    "\n",
    "plink2 --bfile GP2-v4-AAC-noRelateds \\\n",
    "--glm --chr 1 --from-bp 155235878 --to-bp 155235878 \\\n",
    "--covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 --out additive_AAC --ci 0.95\n",
    "\n",
    "# CHROM\tPOS\tID\tREF\tALT\tA1\tTEST\tOBS_CT\tOR\tSE\tL95\tU95\tT_STAT\tP\n",
    "# 1\t155235878\tchr1:155235878:G:T\tT\tG\tG\tADD\t1323\t1.95981\t0.146109\t1.47179\t2.60964\t4.60511\t4.12248e-06\n",
    "\n",
    "cd GP2-v4-AAC\n",
    "\n",
    "plink2 --bfile GP2-v4-AAC-noRelateds --glm recessive --chr 1 --from-bp 155235878 --to-bp 155235878 \\\n",
    "--covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--out recessive_AAC --ci 0.95\n",
    "\n",
    "# CHROM\tPOS\tID\tREF\tALT\tA1\tTEST\tOBS_CT\tOR\tSE\tL95\tU95\tT_STAT\tP\n",
    "# 1\t155235878\tchr1:155235878:G:T\tT\tG\tG\tREC\t1323\t4.38998\t0.40664\t1.97847\t9.74083\t3.63792\t0.000274846\n",
    "```"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "476c8a17-114c-4e99-a3e9-83a0ec45f8c0",
   "metadata": {},
   "source": [
    "```bash\n",
    "####### AFR #######\n",
    "\n",
    "cd AFR/GP2-v4-AFR-wNIGERIAN-NB\n",
    "\n",
    "plink2 --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--glm dominant --chr 1 --from-bp 155235878 --to-bp 155235878 \\\n",
    "--covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--out dominant_AFR --ci 0.95\n",
    "\n",
    "# CHROM\tPOS\tID\tREF\tALT\tA1\tTEST\tOBS_CT\tOR\tSE\tL95\tU95\tT_STAT\tP\n",
    "# 1\t155235878\tchr1:155235878:G:T\tT\tG\tG\tDOM\t1673\t1.74201\t0.108915\t1.40716\t2.15655\t5.09611\t3.46711e-07\n",
    "\n",
    "cd AFR/GP2-v4-AFR-wNIGERIAN-NB\n",
    "\n",
    "plink2 --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--glm --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--out additive_AFR --ci 0.95\n",
    "\n",
    "# CHROM\tPOS\tID\tREF\tALT\tA1\tTEST\tOBS_CT\tOR\tSE\tL95\tU95\tT_STAT\tP\n",
    "# 1\t155235878\tchr1:155235878:G:T\tT\tG\tG\tADD\t1673\t1.75163\t0.0873806\t1.47592\t2.07884\t6.41499\t1.40836e-10\n",
    "\n",
    "cd AFR/GP2-v4-AFR-wNIGERIAN-NB\n",
    "\n",
    "plink2 --bfile GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--glm recessive --chr 1 --from-bp 155235878 --to-bp 155235878 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--out recessive_AFR --ci 0.95\n",
    "\n",
    "# CHROM\tPOS\tID\tREF\tALT\tA1\tTEST\tOBS_CT\tOR\tSE\tL95\tU95\tT_STAT\tP\n",
    "# 1\t155235878\tchr1:155235878:G:T\tT\tG\tG\tREC\t1673\t3.46287\t0.223441\t2.23482\t5.36573\t5.55895\t2.71399e-08\n",
    "```"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "a69dbee0-6f7a-4244-a18d-7cd0b78acb73",
   "metadata": {},
   "source": [
    "```bash\n",
    "### 1_155235878 AAO analyses in the AAC and AFR cohorts + meta-analysis\n",
    "\n",
    "cd AAC/GP2-v4-AAC\n",
    "\n",
    "plink --bfile Cases_GP2-v4-AAC-noRelateds \\\n",
    "--chr 1 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--from-bp 155235878 --to-bp 155235878 \\\n",
    "--linear --out linear_regression \\\n",
    "--pheno AFR/GP2-v4-AFR-wNIGERIAN-NB/age_cases.txt\n",
    "\n",
    "# cat linear_regression.assoc.linear\n",
    "#  CHR                  SNP         BP   A1       TEST    NMISS       BETA         STAT            P \n",
    "#    1   chr1:155235878:G:T  155235878    G        ADD      183     -4.156       -2.443      0.01559\n",
    "#    1   chr1:155235878:G:T  155235878    G        SEX      183     0.6536       0.3176       0.7512\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC1      183    0.02188       0.1637       0.8701\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC2      183    -0.1008      -0.1855        0.853\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC3      183     -0.256      -0.7346       0.4636\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC4      183    -0.7727       -1.502        0.135\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC5      183     0.2459       0.5869        0.558\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC6      183     -1.119       -1.735      0.08456\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC7      183    -0.7827       -1.549       0.1232\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC8      183     0.2335       0.5359       0.5927\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC9      183     0.3095       0.5127       0.6088\n",
    "#    1   chr1:155235878:G:T  155235878    G       PC10      183    -0.7813       -1.406       0.1615\n",
    "\n",
    "cd AFR/GP2-v4-AFR-wNIGERIAN-NB\n",
    "\n",
    "plink --bfile Cases_GP2-v4-AFR-wNIGERIAN-NB-noRelateds \\\n",
    "--chr 1 --covar masterfile_updated_GP2_v4_covariateFile_wAGE_FEB2023.txt \\\n",
    "--covar-name SEX,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10 \\\n",
    "--from-bp 155235878 --to-bp 155235878 \\\n",
    "--linear --out linear_regression \\\n",
    "--pheno age_cases.txt \n",
    "\n",
    "# cat linear_regression.assoc.linear\n",
    "#  CHR                  SNP         BP   A1       TEST    NMISS       BETA         STAT            P \n",
    "#    1   chr1:155235878:G:T  155235878    G        ADD      687     -2.004       -3.483    0.0005284\n",
    "#    1   chr1:155235878:G:T  155235878    G        SEX      687    -0.6066      -0.6941       0.4878\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC1      687    -0.1977       -1.083        0.279\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC2      687     0.1228       0.4397       0.6603\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC3      687    -0.1012      -0.3755       0.7074\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC4      687    -0.3303       -1.219       0.2233\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC5      687    -0.5364       -2.115      0.03476\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC6      687    -0.3715       -1.391       0.1646\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC7      687     -0.282       -1.178       0.2393\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC8      687    -0.2463       -1.448       0.1481\n",
    "#    1   chr1:155235878:G:T  155235878    G        PC9      687     0.2625        1.022        0.307\n",
    "#    1   chr1:155235878:G:T  155235878    G       PC10      687     0.3814        1.905      0.05719\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9aec486-7d2d-4178-bc9c-df613919cb11",
   "metadata": {},
   "source": [
    "```R\n",
    "install.packages(\"rmeta\")\n",
    "library(rmeta)\n",
    "library(data.table)\n",
    "sumstats <- fread(\"summary.txt\", header = T)\n",
    "met <- meta.summaries(d = sumstats$BETA, se = sumstats$SE, method = c(\"fixed\"), logscale = F, names = sumstats$COHORT) # logscale option set for OR.\n",
    "\n",
    "met$test # gives you the Z and P for the meta-analysis.\n",
    "met\n",
    "\n",
    "## Make summary.txt table\n",
    "# BETA\tSE\tP\tCOHORT\n",
    "# -2.004\t0.57\t0.0005\tAfrican\n",
    "# -4.156\t0.58\t0.015\tAfricanAdmixed\n",
    "\n",
    "# Fixed-effects meta-analysis\n",
    "# Call: meta.summaries(d = sumstats$BETA, se = sumstats$SE, method = c(\"fixed\"), \n",
    "#     logscale = F, names = sumstats$COHORT)\n",
    "# Summary effect=-3.06   95% CI (-3.86, -2.26)\n",
    "# Estimated heterogeneity variance: 2  p= 0.008 \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93a9b29d-3527-4cda-8b8d-d4287ca10afe",
   "metadata": {},
   "source": [
    "# WGS Annotation\n",
    "For T2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5353085c-b12e-4d71-85ee-7d7311ff7ee1",
   "metadata": {},
   "source": [
    "```bash\n",
    "## Annotate GBA in WGS data (206 samples)\n",
    "cd /data/CARD_training\n",
    "module load annovar\n",
    "## test.vcf contains WGS data only for GBA \n",
    "\n",
    "table_annovar.pl test.vcf $ANNOVAR_DATA/hg38 -buildver hg38 \\\n",
    "--thread 16 \\\n",
    "--out ALL.annovar \\\n",
    "-arg '-splicing 15',,, \\\n",
    "-remove \\\n",
    "-protocol refGene,avsnp147,ljb26_all,gnomad312_genome \\\n",
    "-operation g,f,f,f \\\n",
    "-nastring . \\\n",
    "--vcfinput\n",
    "\n",
    "## Now extract carriers for coding variants \n",
    "\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155236249 --to-bp 155236249 --recode A --pheno pheno.txt --out p.I320S --double-id\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155236269 --to-bp 155236269 --recode A --pheno pheno.txt --out p.M313I --double-id\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155236377 --to-bp 155236377 --recode A --pheno pheno.txt --out p.G277G --double-id\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155238228 --to-bp 155238228 --recode A --pheno pheno.txt --out p.W136R --double-id\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155239962 --to-bp 155239962 --recode A --pheno pheno.txt --out p.S77R --double-id\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155240707 --to-bp 155240707 --recode A --pheno pheno.txt --out p.K13R --double-id\n",
    "\n",
    "## Confirm top hit carriers\n",
    "\n",
    "plink --vcf test.vcf --chr 1 --from-bp 155235878 --to-bp 155235878 --recode A --pheno pheno.txt --out tophit_155235878 --double-id\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2e5475e-4d94-4af7-bfbf-6a7f78220c2b",
   "metadata": {},
   "source": [
    "# Conditional Analysis"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "89236684-6397-4113-ba8d-116e1f70f159",
   "metadata": {},
   "source": [
    "```bash\n",
    "###### CONDITIONAL ANALYSES on the top two SNPs ###### \n",
    "\n",
    "## NIGERIA 2\n",
    "\n",
    "cat list | while read LINE \n",
    "do\n",
    "echo $LINE\n",
    "plink2 --pfile chr1_AFRICAN --extract range GBA_SNPS.txt --make-bed --out $LINE.AFRICAN\n",
    "done\n",
    "\n",
    "plink --bfile 1_155869005.AFRICAN --merge-list list_SNPS.txt --make-bed --out ALL_35SNPS.AFRICAN\n",
    "\n",
    "plink --bfile ALL_35SNPS.AFRICAN --pheno pheno_AFRICA.txt --update-sex sex_AFRICA.txt --recode A --out ALL_35SNPS.AFRICAN_pheno_sex\n",
    "\n",
    "## AAC\n",
    "\n",
    "cat list | while read LINE \n",
    "do\n",
    "echo $LINE\n",
    "plink2 --pfile chr1_AAC --extract range GBA_SNPS.txt --make-bed --out $LINE.AAC\n",
    "done\n",
    "\n",
    "plink --bfile 1_155869005.AAC --merge-list list_SNPS.txt --recode A --out ALL_35SNPS.AAC\n",
    "\n",
    "## REFORMAT\n",
    "\n",
    "sed \"s/:/_/g\" ALL_35SNPS.AAC.raw > ALL_35SNPS.AAC_reformatted.raw\n",
    "grep -v \"IPDGCAF\" ALL_35SNPS.AAC_reformatted.raw > ALL_35SNPS.AAC_reformatted_final.raw\n",
    "\n",
    "sed \"s/:/_/g\" ALL_35SNPS.AFRICAN_pheno_sex.raw > ALL_35SNPS.AFRICAN_reformatted.raw\n",
    "\n",
    "## CHECK R2\n",
    "```"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "75a13171-c9aa-48f1-841c-c4a5492460f8",
   "metadata": {},
   "source": [
    "```R\n",
    "# Data Munging\n",
    "\n",
    "library(data.table)\n",
    "cov_df <- fread(\"covariate_file_wAGE.txt\", header = T)\n",
    "snps_df <- fread(\"ALL_35SNPS.AFRICAN_reformatted.raw\", header = T)\n",
    "merged_df <- merge(cov_df, snps_df, by=\"IID\")\n",
    "summary(merged_df)\n",
    "merged_df$CASE = merged_df$PHENO - 1\n",
    "merged_df$female = merged_df$SEX.x - 1\n",
    "merged_df$dataset = \"AFR\"\n",
    "\n",
    "# Model parameters\n",
    "\n",
    "snp_params <- paste(names(snps_df)[7:59], collapse=' + ')\n",
    "other_params <- \"CASE ~ female\"\n",
    "pc_params <- \"PC1 + PC2 + PC3 + PC4 + PC5\"\n",
    "model_formula <- paste(other_params, snp_params, pc_params, sep =\" + \")\n",
    "\n",
    "# Build model\n",
    "\n",
    "conditional_model <- glm(model_formula, data = merged_df, family=\"binomial\")\n",
    "summary(conditional_model)\n",
    "\n",
    "# Data Munging\n",
    "\n",
    "library(data.table)\n",
    "cov_df <- fread(\"covariate_file_wAGE.txt\", header = T)\n",
    "snps_df <- fread(\"ALL_35SNPS.AAC_reformatted.raw\", header = T)\n",
    "merged_df <- merge(cov_df, snps_df, by=\"IID\")\n",
    "summary(merged_df)\n",
    "merged_df$CASE = merged_df$PHENO - 1\n",
    "merged_df$female = merged_df$SEX.x - 1\n",
    "merged_df$dataset = \"AFR\"\n",
    "\n",
    "# Model parameters\n",
    "\n",
    "snp_params <- paste(names(snps_df)[7:59], collapse=' + ')\n",
    "other_params <- \"CASE ~ female\"\n",
    "pc_params <- \"PC1 + PC2 + PC3 + PC4 + PC5\"\n",
    "model_formula <- paste(other_params, snp_params, pc_params, sep =\" + \")\n",
    "\n",
    "# Build model\n",
    "\n",
    "conditional_model <- glm(model_formula, data = merged_df, family=\"binomial\")\n",
    "summary(conditional_model)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81808763-08ed-43b9-8eee-bf305a0e3dd3",
   "metadata": {},
   "source": [
    "# Miami plot compared to Europeans"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "fa63586d-dcd4-4ee2-8eac-e9ddffe6e275",
   "metadata": {},
   "source": [
    "```bash\n",
    "cd Miami_Plot_equal_sample\n",
    "\n",
    "## METAL AFR + AAC data without 23andMe Sum stats = 1,200 cases + 2,445 controls\n",
    "\n",
    "# UNCOMMENT THE NEXT LINE TO ENABLE GenomicControl CORRECTION\n",
    "SCHEME STDERR\n",
    "GENOMICCONTROL ON\n",
    "\n",
    "# === DESCRIBE AND PROCESS THE FIRST INPUT FILE ===\n",
    "MARKER markerID\n",
    "ALLELE effect_allele other_allele\n",
    "# FREQ   maf\n",
    "EFFECT beta\n",
    "STDERR LOG(OR)_SE\n",
    "PVALUE P\n",
    "# WEIGHT OBS_CT \n",
    "PROCESS GP2-v4-AAC-GWAS-MAF005-FEB2023.UpdatedforMETAL.tab\n",
    "\n",
    "# === DESCRIBE AND PROCESS THE SECOND INPUT FILE ===\n",
    "MARKER MarkerName\n",
    "ALLELE Allele1\tAllele2\n",
    "# FREQ   maf\n",
    "EFFECT Effect\n",
    "STDERR StdErr\n",
    "PVALUE P-value\n",
    "# WEIGHT N\n",
    "PROCESS AFR-ONLY-META-UpdatedforMETAL1.tbl\n",
    "\n",
    "\n",
    "OUTFILE MY_TWO_GWAS_FILES_META.tbl\n",
    "ANALYZE HETEROGENEITY\n",
    "\n",
    "QUIT\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b453a36d-8465-4690-b916-a793cc6cee7c",
   "metadata": {},
   "source": [
    "```R\n",
    "## Format sum stats splitting chr and pos\n",
    "\n",
    "library(data.table)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(plyr)\n",
    "META_AFR_AAC <- fread(\"METAANALYSIS1.TBL\", header = T)\n",
    "META_AFR_AAC$CHR <- ldply(strsplit(as.character(META_AFR_AAC$MarkerName), split = \":\"))[[1]]\n",
    "META_AFR_AAC$BP <- ldply(strsplit(as.character(META_AFR_AAC$MarkerName), split = \":\"))[[2]]\n",
    "write.table(META_AFR_AAC, file = \"chr_pos_METAANALYSIS1.TBL\", quote = F, row.names = F, sep = \"\\t\")\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd278727-6691-4047-a940-af2ad472bb95",
   "metadata": {},
   "source": [
    "```bash \n",
    "sed 's/chr//g' chr_pos_METAANALYSIS1.TBL > formatted_chr_pos_METAANALYSIS1.TBL\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e668515d-f0c3-436b-9e68-a307efeac384",
   "metadata": {},
   "source": [
    "```bash\n",
    "## EUR GWAS at similar N cases and controls\n",
    "\n",
    "## Randomly select cases\n",
    "cd /data/CARD/PD/GP2/genotypes/GP2/round3/imputed/EUR\n",
    "\n",
    "module load plink/2.0_alpha_1_final\n",
    "plink2 --pfile /data/CARD/PD/GP2/genotypes/GP2/round3/imputed/EUR/chr9_EUR_prune_final \\\n",
    "--make-bed --out chr9_EUR_prune_final\n",
    "\n",
    "module load plink/1.9\n",
    "plink --bfile chr9_EUR_prune_final \\\n",
    "--filter-cases --out cases --make-bed\n",
    "\n",
    "## Here we have 7,409 cases and we want to keep 1,200 \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7bdea11d-a2ff-403c-94d4-7d380de501d3",
   "metadata": {},
   "source": [
    "```R\n",
    "\n",
    "library(data.table)\n",
    "fam <- fread(\"cases.fam\", header = F)\n",
    "colnames(fam) <- c(\"FID\", \"IID\", \"MAT\", \"PAT\", \"SEX\", \"PHENO\")\n",
    "fam_subsetted1 <- fam[sample(1:nrow(fam), 1200), ]\n",
    "write.table(fam_subsetted1, \"EUR_tokeep_cases.txt\", quote = F, sep = \"\\t\", row.names = F, col.names = F)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1159290c-afbe-44d4-9ecb-ff90a9418990",
   "metadata": {},
   "source": [
    "```bash\n",
    "## Randomly select controls\n",
    "\n",
    "module load plink/1.9\n",
    "plink --bfile chr9_EUR_prune_final --filter-controls --out controls --make-bed\n",
    "\n",
    "## Here we have 4,277 controls and we want to keep 2,445 \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e965653c-d99b-4d02-9b92-388682d7f01f",
   "metadata": {},
   "source": [
    "```R\n",
    "\n",
    "library(data.table)\n",
    "fam <- fread(\"controls.fam\", header = F)\n",
    "colnames(fam) <- c(\"FID\", \"IID\", \"MAT\", \"PAT\", \"SEX\", \"PHENO\")\n",
    "fam_subsetted1 <- fam[sample(1:nrow(fam), 2445), ]\n",
    "write.table(fam_subsetted1, \"EUR_tokeep_controls.txt\", quote = F, sep = \"\\t\", row.names = F, col.names = F)\n",
    "```"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "d14032d8-84a9-4cac-83cf-a172e5b5880d",
   "metadata": {},
   "source": [
    "```bash\n",
    "## Merge list of cases and controls\n",
    "cat EUR_tokeep_cases.txt EUR_tokeep_controls.txt > ALL_tokeep.txt\n",
    "awk '{print $1, $2}' ALL_tokeep.txt> IDs_ALL_tokeep.txt\n",
    "ALL_tokeep.txt IDs_formatted_to_keep.txt > pheno_test.txt\n",
    "awk '{print $7, $8, $6}' pheno_test.txt > pheno.txt\n",
    "\n",
    "## Extract cases and controls and run GWAS\n",
    "module load plink/2.0_alpha_1_final\n",
    "cat list.txt | while read LINE\n",
    "do\n",
    "echo $LINE\n",
    "plink2 --pfile ${LINE}_EUR \\\n",
    "--keep IDs_formatted_to_keep.txt \\\n",
    "--pheno pheno.txt --make-bed \\\n",
    "--out Miami_Plot_equal_sample/subsetted${LINE}\n",
    "done \n",
    "\n",
    "module load plink/1.9\n",
    "cat list.txt | while read LINE\n",
    "do\n",
    "echo $LINE\n",
    "plink --bfile Miami_Plot_equal_sample/subsetted${LINE} \\\n",
    "--allow-no-sex \\\n",
    "--logistic --out Miami_Plot_equal_sample/sampled$LINE\n",
    "done \n",
    "\n",
    "cat sampledchr*.assoc.logistic > ALL_EUR_assoc.logistic\n",
    "grep -v \"TEST\" ALL_EUR_assoc.logistic > test_ALL_EUR_assoc.logistic\n",
    "grep -v \"NA\" test_ALL_EUR_assoc.logistic > no_NA_test_ALL_EUR_assoc.logistic\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ced089a9-781d-4737-9556-96a8924e735d",
   "metadata": {},
   "source": [
    "```R\n",
    "library(data.table)\n",
    "stats <- fread(\"no_NA_test_ALL_EUR_assoc.logistic\", header = F)\n",
    "colnames(stats) <- c(\"CHR\",\"SNP\",\"BP\",\"A1\",\"TEST\",\"NMISS\",\"OR\",\"STAT\",\"P\")\n",
    "subset(stats, P < 1E-100)\n",
    "write.table(stats_revised, \"ALL_EUR_assoc_header.logistic\", quote = F, sep = \"\\t\", row.names = F, col.names = T)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc008c54-7e1d-4fe6-8154-f6482ad264ba",
   "metadata": {},
   "source": [
    "```python\n",
    "## Miami_plot\n",
    "import gwaslab as gl\n",
    "import pandas as pd\n",
    "\n",
    "mysumstats = gl.plot_miami(path1=\"formatted_chr_pos_METAANALYSIS1.TBL\",\n",
    "\t\t\t\t\t\t   path2=\"ALL_EUR_assoc_header.logistic\",\n",
    "                           cols1=[\"CHR\",\"BP\",\"P-value\"],\n",
    "                           cols2=[\"CHR\",\"BP\",\"P\"],\n",
    "                           titles=[\"African & African Admixed GWAS meta-analysis\", \"European GWAS meta-analysis\"],\n",
    "                           titles_pad=[0.15,0.0],\n",
    "                           cut=20,\n",
    "                           region_grid=True,\n",
    "                           save=\"Miami_sampled.jpg\",\n",
    "                           saveargs={\"dpi\":400,\"facecolor\":\"white\"}\n",
    "                           )\n",
    "```"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "28abf3e0",
   "metadata": {},
   "source": [
    "# GBA Gauchian Caller"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "b3d47265",
   "metadata": {},
   "source": [
    "```wdl\n",
    "version 1.0\n",
    "# WORKFLOW DEFINITION \n",
    "\n",
    "workflow CallGBAvariants {\n",
    "  input {\n",
    "    File cram\n",
    "    File cram_index\n",
    "    File ref_fasta\n",
    "    File ref_fasta_index\n",
    "    String sample_id\n",
    "    String? gauch_docker_override\n",
    "    String gauch_docker = select_first([gauch_docker_override, \"avalillarionova/gauchian:1.0.2\"])\n",
    "    Int? preemptible_attempts_override\n",
    "    Int preemptible_attempts = select_first([preemptible_attempts_override, \"2\"])\n",
    "    String runtime_zones\n",
    "    Int threads\n",
    "    }\n",
    "\n",
    "  call Gauchian {\n",
    "      input:\n",
    "        cram = cram,\n",
    "        cram_index= cram_index,\n",
    "        ref_fasta = ref_fasta,\n",
    "        ref_fasta_index = ref_fasta_index,\n",
    "        sample_id = sample_id,\n",
    "        gauch_docker = gauch_docker,\n",
    "\t    threads = threads,\n",
    "        preemptible_attempts = preemptible_attempts,\n",
    "        runtime_zones = runtime_zones\n",
    "    }\n",
    "      # Outputs that will be retained when execution is complete\n",
    "  output {\n",
    "      File gauchian_tsv = Gauchian.output_tsv\n",
    "      File gauchian_json = Gauchian.output_json\n",
    "  }\n",
    "\n",
    "}\n",
    "\n",
    "# TASK DEFINITIONS\n",
    "\n",
    "task Gauchian {\n",
    "  input {\n",
    "  # Command parameters\n",
    "    File cram\n",
    "    File cram_index\n",
    "    File ref_fasta\n",
    "    File ref_fasta_index\n",
    "    String sample_id\n",
    "\n",
    "  # Runtime parameters\n",
    "    Int additional_disk_space_gb = 10\n",
    "    Int? machine_mem_gb\n",
    "    Int preemptible_attempts\n",
    "    String gauch_docker\n",
    "    String runtime_zones\n",
    "    Int threads  \n",
    "    Int disk_space_gb = ceil(2*size(cram, \"G\") + size(ref_fasta, \"G\")) + additional_disk_space_gb\n",
    "  }\n",
    "\n",
    "  command {  \n",
    "  echo \"~{cram}\" > manifest_gauch.txt\n",
    "  python -m gauchian --manifest manifest_gauch.txt \\\n",
    "                   --genome 38 \\\n",
    "                   --reference ~{ref_fasta} \\\n",
    "                   --prefix ~{sample_id} \\\n",
    "                   --outDir ~{sample_id} \\\n",
    "                   --threads ~{threads}\n",
    "\n",
    "  mv ~{sample_id}/*.tsv \"~{sample_id}.tsv\"\n",
    "  mv ~{sample_id}/*.json \"~{sample_id}.json\"\n",
    "  }\n",
    "  runtime {\n",
    "    docker: gauch_docker\n",
    "    memory: select_first([machine_mem_gb, 10]) + \" GB\"\n",
    "    cpu: threads\n",
    "    disks: \"local-disk \" + disk_space_gb + \" HDD\"\n",
    "    preemptible: preemptible_attempts\n",
    "    zones: runtime_zones\n",
    "  }\n",
    "  output {\n",
    "    File output_tsv = \"~{sample_id}.tsv\"\n",
    "    File output_json = \"~{sample_id}.json\"\n",
    "  }\n",
    "}\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a74c6db",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.9",
   "language": "python",
   "name": "py3.9"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
